src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage.Accessor.cs

```diff
@@ -249,7 +237,8 @@ private bool TryGetRowIdWorker(SqlConnection connection, TDatabaseId dataId, out
                 // ROWID, _ROWID_, or OID. Except if you declare an ordinary table column to use one 
                 // of those special names, then the use of that name will refer to the declared column
                 // not to the internal ROWID.
-                using (var resettableStatement = connection.GetResettableStatement(_select_rowid_from_0_where_1))
+                using (var resettableStatement = connection.GetResettableStatement(
+                    $@"select rowid from {dbName}.{DataTableName} where ""{DataIdColumnName}"" = ?"))
```

Comment:
note: these add string allocations. if this is an issue we can move to a model where we cache these depending on which DB we're reading.

Reply:
Those string allocations do worry me a bit.

In fact, I'm also glaring at the fact that the statement is being re-prepared and finalized on every call.  Caching the prepared statement is often a nice perf win, although it does mean you have to be more careful about how the db connection is closed.  SQLite will not allow you to close a db connection which has un-finalized statements.  (Note that this comment is using the SQLite definition of finalize, not the .NET meaning of same.)

Reply:
> In fact, I'm also glaring at the fact that the statement is being re-prepared and finalized on every call

That's not what happens here.  GetResettableStatement returns a pooled statement for that string to prevent having to reparse or reprepare.

Reply:
>  SQLite will not allow you to close a db connection which has un-finalized statements. (Note that this comment is using the SQLite definition of finalize, not the .NET meaning of same.)

When we dispose our storage service, we go through and dispose all our pooled connection objects.  Our connections then dispose all the cache statements they have.

Reply:
OK, we're good on the prepared statement thing.  Thanks for clarifying.